app/controllers/coaches_controller.rb

```diff
@@ -8,26 +8,26 @@ def new_registration_form
   def create
     @coach = Coach.new
                 
-    @coach.password = params.fetch(:password, nil)
-    @coach.password_confirmation = params.fetch(:password_confirmation, nil)
+    @coach.password = params.fetch(:password)
+    @coach.password_confirmation = params.fetch(:password_confirmation)
                 
-    @coach.email = params.fetch(:email, @coach.email)
+    @coach.email = params.fetch(:email)
                 
-    @coach.first_name = params.fetch(:first_name, @coach.first_name)
+    @coach.first_name = params.fetch(:first_name)
                 
-    @coach.middle_name = params.fetch(:middle_name, @coach.middle_name)
+    @coach.middle_name = params.fetch(:middle_name)
                 
-    @coach.bio = params.fetch(:bio, @coach.bio)
+    @coach.bio = params.fetch(:bio)
                 
-    @coach.a_bool = params.fetch(:a_bool, @coach.a_bool)
+    @coach.a_bool = params.fetch(:a_bool)
```

Comment:
The checkbox is missing it's fallback; this is the one case where we need it.
```suggestion
    @coach.a_bool = params.fetch(:a_bool, false)
```

Reply:
Yes, `account` didn't have any of the field type conditionals that resources has. I've add those here.